File: Data Structures & Algorithms/balanced-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        
        def dfs(root):
            if not root:
                return 0, True
            lh, l_res = dfs(root.left)
            rh, r_res = dfs(root.right)
            mh = max(lh, rh)
            is_balanced = l_res and r_res and abs(lh- rh) <= 1
            return mh + 1, is_balanced
        return dfs(root)[1]

File: Data Structures & Algorithms/is-anagram/submission-2.js
class Solution {
    /**
     * @param {string} s
     * @param {string} t
     * @return {boolean}
     */
    isAnagram(s, t) {
        if (s.length !== t.length) {
            return false;
        }
        const left = {};
        const right = {};
        for (let s_item of s) {
            if (s_item in left) {
                left[s_item] += 1;
            } else {
                left[s_item] = 0;
            }
        }
        
        for (let t_item of t) {
            if (t_item in right) {
                right[t_item] += 1;
            } else {
                right[t_item] = 0;
            }
        }

        for (const key in right) {
            if (left[key] !== right[key]) {
                return false;
            }
        }
        return true;
    }
}
